File: BOJ/python/15947.py
# 노래 가사를 list에 넣어주었다.
song = [
    "baby", "sukhwan", "tururu", "turu", "very", "cute", "tururu", "turu", 
    "in", "bed", "tururu", "turu", "baby", "sukhwan"
]

n = int(input())
checkTuru = n%14 # 현재 index가 어느 word를 가리키는지 확인
verse = n//14 # 노래가 몇 절인지 확인
if checkTuru == 3 or checkTuru == 7 or checkTuru == 11: # 1절 기준 'tururu'위치
    if verse >= 3: # 3절 이상일 경우 'ru'가 5번 이상 등장하므로 'tu+ru*5'와 같은 형식으로 출력
        print(f"tu+ru*{verse+2}")
    else: # 아닐경우 turururu와 같이 'ru'를 이어붙여 출력
        print("tururu"+"ru"*verse)
elif checkTuru == 4 or checkTuru == 8 or checkTuru == 12: # 1절 기준 'turu'위치
    if verse >= 4: # 4절 이상일 경우 'ru'가 5번 이상 등장하므로 'tu+ru*5'와 같은 형식으로 출력
        print(f"tu+ru*{verse+1}")
    else: # 아닐 경우 'ru'를 이어붙여 출력
        print("turu"+"ru"*verse)
else: # 1절을 기준으로 'tururu' or 'turu'의 위치가 아니라면 해당 word를 출력
    print(f"{song[checkTuru-1]}")

File: BOJ/15947.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	string song[15] = { "baby", "sukhwan", "tururu", "turu", "very", "cute", "tururu", "turu",
		"in", "bed", "tururu", "turu", "baby", "sukhwan" };
	int n;
	cin >> n;
	n--;

	int checkTuru = n % 14; // tururu, turu check
	int verse = n / 14; // verse check

	if (song[checkTuru] == "tururu") { // 1절을 기준으로 tururu 위치일 경우
		if (verse >= 3) cout << "tu+ru*" << verse + 2; // 3절 이상일 경우 ru의 등장 횟수가 5이상이다.
		else {
			cout << "tururu";
			for (int i = 1; i <= verse; i++) cout << "ru";
		}
	}
	else if (song[checkTuru] == "turu") { // 1절을 기준으로 turu 위치일 경우
		if (verse >= 4) { // 4절 이상일 경우 ru의 등장 횟수가 5 이상이다.
			cout << "tu+ru*" << verse + 1;
		}
		else {
			cout << "turu";
			for (int i = 1; i <= verse; i++) cout << "ru";
		}
	}
	else cout << song[checkTuru]; // tururu, turu가 아닐 경우 해당 word를 출력

	return 0;
}